Schedule preview improvements

ryan-conn opened the pull request:
## Description

Replaces color boxes for sections with no meeting times with an X, and shows instructor names in the preview

## Rationale
I based this branch off of `schedule-details` because I reorganized the schedule preview significantly in that branch. No tests because I don't think they'd be very helpful but I could try to add some if we want.

I also resized `<ColorBox>` to be 1px bigger because the old size made it impossible to center the X, it would always be 1px off because of rounding

## How to test
Mess around with adding async classes, also try resizing the window to see how the instructor names interact when cut off (they should truncate the text and add an ellipsis)

## Screenshots

Before|After
--|--
![image](https://user-images.githubusercontent.com/28655462/106398645-b1626200-63d9-11eb-8638-7ef5af5608ab.png)|![image](https://user-images.githubusercontent.com/28655462/107126159-dd676280-6873-11eb-80fb-19093d3935d2.png)


## Related tasks

Closes #491, #297.

gannonprudhomme commented on autoscheduler/frontend/src/components/SchedulingPage/SchedulePreview/ScheduleListItem/ScheduleListItem.tsx:
```diff
@@ -59,13 +59,23 @@ const ScheduleListItem: React.FC<ScheduleListItemProps> = ({ index, onDetailsCli
-    </span>
-  ));
+  const scheduleSections = sectionsForSchedule(schedule).map((sec: Section) => {
+    const color = !sec.asynchronous && meetingColors.get(sec.subject + sec.courseNum);
```
This is like an alternative form of a ternary where the false case is undefined/null right?

ryan-conn replied:
Yeah although I probably should've just explicitly set it to undefined to make it more clear

gannonprudhomme replied:
Yeah I'd say so, honestly I don't really like this syntax that much in this scenario (I'm more inclined to favor it when it's in actual JSX)

gannonprudhomme reviewed (approved):
This is pretty straightforward and I'll approve it since I think it's mostly good and achieves what it's supposed to (the red X looks much better too), however:

- I'd honestly wait to merge this into schedule details and instead just merge it into master once schedule details is merged, but up to you (I wish I did that for term select and last updated indicator since they're loosely related but technically separate tbh)
- Honestly I think I still would like the instructors to be left-aligned with each other instead of being slightly offset. I think not having them aligned makes the schedule preview feel way more cluttered/disorganized, but ik y'all said otherwise last time we discussed this, so if you _really_ disagree then you can leave it (I do think it's work making a demo of it and showing it tho)

ryan-conn commented:
Updated with the updated schedule details PR and addressed Gannon's comments


> I'd honestly wait to merge this into schedule details and instead just merge it into master once schedule details is merged, but up to you (I wish I did that for term select and last updated indicator since they're loosely related but technically separate tbh)

Just for clarity's sake this is what I decided to do

gannonprudhomme commented:
Think you need to hard reset and cherry pick these

ryan-conn commented:
> Think you need to hard reset and cherry pick these

Correct, I really didn't want to deal with 100+ lines of merge conflicts so that's what I ended up doing

Conveniently I noticed that the sections for schedules were throwing errors because I forgot to give them keys, so I fixed that as well. Other than that this should be identical to before I cherry-picked
